ladybug/dt.py

```diff
@@ -220,3 +224,199 @@ def ToString(self):
         return self.__str__()
+
+
+class Date(date):
+    """Ladybug Date.
+
+    Properties:
+        month
+        day
+        leap_year
+        doy
+    """
+
+    __slots__ = ()
+
+    def __new__(cls, month=1, day=1, leap_year=False):
+        """Create Ladybug Date.
+
+        Args:
+            month: A value for month between 1-12 (Defualt: 1).
+            day: A value for day between 1-31 (Defualt: 1).
+            leap_year: A boolean to indicate if date is for a leap year
+                (Default: False).
+        """
+        year = 2016 if leap_year else 2017
+        try:
+            return date.__new__(cls, year, month, day)
+        except ValueError as e:
+            raise ValueError("{}:\n\t({}/{})(m/d)".format(e, month, day))
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create date from a dictionary.
+
+        Args:
+            data: {
+                'month': A value for month between 1-12. (Defualt: 1)
+                'day': A value for day between 1-31. (Defualt: 1)
+            }
+        """
+        month = data['month'] if 'month' in data else 1
+        day = data['day'] if 'day' in data else 1
+        leap_year = data['leap_year'] if 'leap_year' in data else False
+        return cls(month, day, leap_year)
+
+    @classmethod
+    def from_doy(cls, doy, leap_year=False):
+        """Create Ladybug Date from an day of the year.
+
+        Args:
+            doy: An int value 0 <= and < 366
+        """
+        if not leap_year:
+            days_until_month = (0, 31, 59, 90, 120, 151, 181,
+                                212, 243, 273, 304, 334, 365)
+        else:
+            days_until_month = (0, 31, 60, 91, 121, 152, 182,
+                                213, 244, 274, 305, 335, 366)
+
+        for month_count in range(12):  # find month
+            if int(doy) < days_until_month[month_count + 1]:
+                month = month_count + 1
+                break
+        try:
+            day = int(doy - days_until_month[month - 1])
+        except UnboundLocalError:
+            raise ValueError(
+                "doy must be positive and smaller than 366. Invalid input %d" % (doy)
+            )
+        return cls(month, day, leap_year)
+
+    @classmethod
+    def from_date_string(cls, date_string, leap_year=False):
+        """Create Ladybug Date from a Date string.
+
+        Usage:
+
+            dt = Date.from_date_string("31 Dec")
+        """
+        dt = datetime.strptime(date_string, '%d %b')
+        return cls(dt.month, dt.day, leap_year)
+
+    @property
+    def leap_year(self):
+        """Boolean to note whether Date belongs to a leap year or not."""
+        return self.year == 2016
+
+    @property
+    def doy(self):
+        """Calculate day of the year for this date."""
+        return self.timetuple().tm_yday
+
+    def to_dict(self):
+        """Get date as a dictionary."""
+        base = {'month': self.month, 'day': self.day}
```

Comment:
are we going to include `type` key?

Reply:
I am thinking that we should but this is an issue much like the documentation one that you pointed out.  Currently, the `type` key isn't used across all of ladybug-core right now and I think it will be better to update it altogether.  So, again, I will leave this as it is now so that we don't get confused about whether we fixed this repo or not.  I have also opened an issue so that we don't forget about it:
https://github.com/ladybug-tools/ladybug/issues/157